### PR TITLE
Sc 847 Task remove not needed safety check from buy and sell in

### PR DESCRIPTION
**Pull request by FHieser**
Removing an unnecessary check from the RedeemingBondingCurveBase
The removed code checks if enough collateral tokens are stored in the contract to do a sell operation.
This is useless in case of a transfer for example as the transfer would fail anyway in case
that there are not enough tokens in the contract.
With this removal we leave it to the PaymentProcessorLogic to fail/handle in case there is not enough collateral to transfer

### CODE REVIEW

File: src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol
Comment (Zitzak) [inline src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol:244]: How would this be handled in case a project is not collecting the project fee directly but storing it in the contract. I.e. the redeem amount might be taken from the project collateral fee stored in the contract

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (Zitzak) [inline src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol]:
```suggestion
 * @notice  Interface that enables the management of the the redemption of
```

**Review by Zitzak — CHANGES_REQUESTED** [review]

File: src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol
Comment (FHieser) [reply src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol:244]: I would say that lies in the responsibility of the functions that are able to transfer collateral tokens out of the contract. In this case the _handleCollateralTokensAfterSell and transferOrchestratorToken in case its a fundingmanager.
The problem is that I dont see a clear way to enforce this properly.

Comment (FHieser) [reply src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol:244]: I had a talk with @marvinkruse about this.
We agreed to enforce the rule of "there has to be enough funds in the BondingCurve to enable the withdrawal of Project fees". This makes the check viable again (without the redeem amount of course).
In the future we probably still have to talk about the Fee System as a whole, as this was a bigger discussion point overall

**Comment by FHieser** [comment]
@Zitzak could you rereview pls?

File: src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol
Comment (Zitzak) [reply src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol:244]: Makes sense. However the way it is implemented atm isn't right I think. Let's do a quick sync around this tomorrow @FHieser.

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol]:
```suggestion
    /// @notice Insufficient collateral tokens are held to cover the project collateral fee.
```

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol]: Needs to be changed once `dev` is updated after the issue in [PR#700](https://github.com/InverterNetwork/contracts/pull/700). Will probably be higher then.

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol]: can be shorted to 
`Module__RedeemingBondingCurveBase__InsufficientCollateralForProjectFee`, the second ´Collateral` is not adding any information as it's clear that is about the collateral token and the name is already quite long?

File: test/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.t.sol
Comment (marvinkruse) [inline test/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.t.sol]:
```suggestion
        // Calculate fee amount
```

File: test/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.t.sol
Comment (marvinkruse) [inline test/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.t.sol]:
```suggestion
        // We simulate the fundingManager spending some funds. It can't cover full redemption anymore.
```

File: src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol]:
```suggestion
        // Require that enough collateral tokens are held to cover the project collateral fee
```

File: src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol:5]: The word "Interface" should not be here, it's the interface of the contract, and the name of the contract is "Inverter Bancor Virtual Supply Bonding Curve Funding Manager", not "Inverter Bancor Virtual Supply Bonding Curve Funding Manager **Interface**".
```suggestion
 * @title   Inverter Bancor Virtual Supply Bonding Curve Funding Manager
```

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol:5]:
```suggestion
 * @title   Inverter Redeeming Bonding Curve Funding Manager Base
```

File: src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol]: Needs to be changed once `dev` is updated after the issue in [PR#700](https://github.com/InverterNetwork/contracts/pull/700). Will definitely be higher then.

**Comment by marvinkruse** [comment]
As mentioned in the other PR: please do not merge this unless I reviewed it, at least until we are all fully used to the Inverter Standard. I'll make sure to quickly review once I'm asked to look at it.

**Review by marvinkruse — CHANGES_REQUESTED** [review]
As mentioned in the other PR: please do not merge this unless I reviewed it, at least until we are all fully used to the Inverter Standard. I'll make sure to quickly review once I'm asked to look at it.

File: src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol
Comment (FHieser) [reply src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol:5]: How would you name an Interface contract then?
When the description is the same as the original contract that it tries to represent wont that be confusing?

File: src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol
Comment (FHieser) [reply src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol]: Yeah
I thought about this too. Concurrent Updates will probably be confusing as hell
Thats where a CI check would be super nice.

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (FHieser) [reply src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol:5]: See other comment

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (FHieser) [reply src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol]: See other comment

**Review by HEBERGEE1 — CHANGES_REQUESTED** [review]
<comment deleted by admin - spam>

File: src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.sol]:
```suggestion
        // Require that enough collateral tokens are held to cover the project
        // collateral fee.
```

File: test/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.t.sol
Comment (marvinkruse) [inline test/modules/fundingManager/bondingCurve/abstracts/RedeemingBondingCurveBase_v1.t.sol]:
```suggestion
        // We simulate the fundingManager spending some funds. 
        // It can't cover full redemption anymore.
```

**Review by marvinkruse — CHANGES_REQUESTED** [review]

**Review by marvinkruse — APPROVED** [review]

File: src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol]:
```suggestion
 * @custom:version 1.1.2
```

Comment (marvinkruse) [reply src/modules/fundingManager/bondingCurve/interfaces/IFM_BC_Bancor_Redeeming_VirtualSupply_v1.sol]: Contract is 1.1.2, thats why the change

File: src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol
Comment (marvinkruse) [inline src/modules/fundingManager/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol]:
```suggestion
 * @custom:version 1.1.3
```

in line with contract.

**Comment by marvinkruse** [comment]
Can be merged from my side. Please remember to choose a good title for the commit and not just take the PRs name. 
For example something like `Fix: Remove redeemed amount from balance check` or so.

**Review by Zitzak — APPROVED** [review]

**Pull request merged**
